dotCMS/src/main/java/com/dotmarketing/portlets/structure/action/EditFieldAction.java

```diff
@@ -318,6 +319,43 @@ private boolean _saveField(ActionForm form, ActionRequest req, ActionResponse re
                  }
             }
 
+            // Validate values entered for decimal/number type check box field
+            if (field.getFieldType().equals(FieldType.RADIO.toString())){
+                String values = fieldForm.getValues();
+                String temp = values.replaceAll("\r\n","|");
+                String[] tempVals = StringUtil.split(temp.trim(), "|");
+
+                if (tempVals.length % 2 == 1){
+                    SessionMessages.add(req, "error", "message.structure.missingdatavalue");
+                    return false;
+                }
+
+                try {
+                    if(dataType.equals(Field.DataType.FLOAT.toString())){
+                        for(int i=1;i<tempVals.length;i+= 2){
+                            Float.parseFloat(tempVals[i]);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The return value of "parseFloat" must be used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2201)